### ISSUE TITLE
'--configuration' option treats casing differently for 'dotnet build' and 'dotnet test' commands

### ISSUE THREAD

**Issue by lGSMl** [dotnet/sdk#15119]
running `dotnet build -c release`
will produce dlls in `<project path>/bin/release/`

but 
running `dotnet test -c release`
will search for test dlls at `<project path>/bin/Release/`, and eventually will fail

**Comment by joeloff** [comment]
@lGSMl Could you provide a bit more details? Are these exunit based tests? I tried this on RHEL with the same SDK by creating a solution with two projects, but running something like ```dotnet build-c releases``` followed by ```dotnet test -c ReLeaSE``` worked. The path appeared to be normalized to bin/Release always.

Are you overriding any properties or using additional SDKs?

**Comment by lGSMl** [comment]
no, it is not exunit - just regular call from shell. I have provided OS details, steps, and docker image where issue occurs and should be reproducible - not sure what other details required

**Comment by lGSMl** [comment]
@joeloff  as I was trying to make a public reproducible scenario I figured out there is indeed a detail I have missed that is a culprit - using RID.

I have created a small showcase with github action that reproduces above mentioned scenario 
https://github.com/lGSMl/dotnet-issue-showcase/runs/1864980214
(never used github actions before - lemme know if I have to grant extra access there or something)

As it is visible - action Build tests runs
`dotnet build -c release --no-restore --verbosity normal --runtime 'linux-musl-x64' ./tests/tests.csproj`
and it creates binaries at `/bin/release`

later step Tests runs 
`dotnet test -c release --no-build --verbosity normal --runtime 'linux-musl-x64'`
and search for binaries at `/bin/Release`

I am updating issue description to include RID part

**Comment by lGSMl** [comment]
going one step further I figured out RID has nothing to do with it - the root cause is that dotnet normalizes `-c release` to `Release` if used on solution, but does not do that if ran on project level. 

The latest showcase can be found here
https://github.com/lGSMl/dotnet-issue-showcase/runs/1865106137

RID topic came up as with latest SDK I can not use RID for the whole solution while building, so I have to build per project, but still can use `dotnet test` on solution level.

Updating issue accordingly

**Comment by joeloff** [comment]
Thanks, this helps a lot. I had access to view the build output.

**Comment by joeloff** [comment]
I'm able to repro this now locally.
